src/Command/InfectionCommand.php

```diff
@@ -86,6 +87,11 @@ final class InfectionCommand extends BaseCommand
      */
     private $skipCoverage;
 
+    /**
+     * @var InfectionContainer|null
```

Comment:
Can it be really `null`?

Reply:
Yes since it is set in `initialize()`

Reply:
Ah, ok, then this is not because it really *can* be null in our application, but because of how symfony commands are being initialized.

In fact, Infection *can not* work without container. It's impossible.

And [this check](https://github.com/infection/infection/pull/759/files#diff-a99db3f318d0547a301f216cb03b51a7R346) is redundant:

```php
$this->container !== null && $this->container->offsetExists($testFileDataProviderServiceId)
```

because container must be initialized.

What I suggest here is to remove `|null`, remove `$this->container &&` conditions and just add `Assert::notNull($this->container)` in the beginning of the `execute()` method if it's required for static analyzers.

Each time I see `Type|null`, I can't use it without checking beforehand, and IDE complains about possible "null pointer exception".

The right way would be rewriting the process how `InfectionCommand` is initialized and pass container in `__construct()`, but this is out of the scope of this PR.

Reply:
>The right way would be rewriting the process how InfectionCommand is initialized and pass container in __construct(), but this is out of the scope of this PR.

Yes that would be the goal.